### ISSUE TITLE
jails directory and images

### ISSUE THREAD

**Issue by krzysztofantczak** [olevole/cbsd#47]
Hi,

I was trying to play around with your preconfigured image (rtorrent). While initializing cbsd i was asked about jails directory and i've selected /jails but after importing rtorrent jail:

```
root@x:/home/x # cbsd jls
rtorrent  0    10.0.0.10/24  rtorrent.x  /usr/jails/jails/rtorrent      Off
root@x:/home/x # cbsd jstart jname=rtorrent
mkdir: /usr/jails/jails: No such file or directory
```

**Comment by krzysztofantczak** [comment]
btw. i saw that your images are really small (like this rtorrent one), do You have any advice to remove not needed files from jail before exporting it to image to make it smaller?

**Comment by olevole** [comment]
Thanks, I will check it. I planned to carry out rework for templates/images for 10.1.5 but I came to a conclusion that their maintenance It is very difficult (for example, software in the images is rapidly becoming to obsolete) in terms of my current employment.  And instead I plan to adapt any software for config management, puppet as example. But am still in meditation on this question ;) it can also greatly affect the size - it will be just the script for CM

**Comment by krzysztofantczak** [comment]
Perhaps ansible instead of puppet? Anyway You mean to use it for bootstrapping new jail while importing it from image? or for configuring it before export?

**Comment by olevole** [comment]
I planned to make it for bootstraping new jail. But the idea of using it also to build and export template image - is  interesting. On the Puppet I'm looking for one reason only - I'm working with him on my daily job and i know Puppet much better. Ansible I used to a lesser extent.

I think an interesting idea, when creating the jails, user can specify a list of the classes from Puppet (or playbook in the case of Ansible or cookbook for chef ;-) to be applied on the jail. If the jail is created via TUI, it can get a list of available classes (playbook) by scanning the predetermined directory and display it in the checkbox form.
It is also interesting to use not only for jail but for bhyve/XEN-based Linux VMs.

**Comment by krzysztofantczak** [comment]
Hmm, also... How can i avoid jails IP conflicts? It seems jail imported from image is using an IP address which was assigned to it before export, is there any way to pick up new IP somehow automatically?

**Comment by olevole** [comment]
Nice idea. Added initial version: https://github.com/olevole/cbsd/commit/029c0606ee61d4745d63d857e1cbeba27df3e45f

~~~
root@home:/ # cbsd jls display=jname,ip4_addr
JNAME         IP4_ADDR
git           192.168.1.50/24
itop          192.168.1.33/24
jail1         192.168.1.3/24
kde4          192.168.1.166/24
kfreebsd1     192.168.1.9/24
pupm          192.168.1.4/24
puppet1       192.168.1.1/24
puppet2       192.168.1.5/24
puppet3       192.168.1.6/24
puppet4       192.168.1.7/24
puppet5       192.168.1.8/24
puppetmaster  192.168.1.44/24

root@home:/ # cbsd jexport jname=puppet1
Images size for puppet1.img: 129m

root@home:/ # cbsd jimport jname=/usr/jails/export/puppet1.img newip=10.0.0.2/24 newjname=test1
IP sets to: 10.0.0.2/24

root@home:/ # cbsd jimport jname=/usr/jails/export/puppet1.img newip=dhcp newjname=test2
IP sets to: 192.168.1.12/24

root@home:/ # cbsd jls display=jname,ip4_addr 
JNAME         IP4_ADDR
..
test1         10.0.0.2/24
test2         192.168.1.12/24
..
~~~

**Comment by krzysztofantczak** [comment]
Cool, so it works also with dhcp? :-D

**Comment by olevole** [comment]
Its fake 'shell-script'-based dhcp ;) Just 'seq 0 255' without any checks for mask or any. As a temporary plug. Location still not fixed: i would like to change the work to an external repository, but at the moment I do not have time for this

**Comment by krzysztofantczak** [comment]
Hmm, do i need to update something more then jimport file? I'm using cbsd-10.1.4_1 (installed with pkgng) and this is my result (after updating this particular file):

```
root@base:/usr/local/cbsd/tools # cbsd jls display=jname,ip4_addr | grep ifconfig
ifconfig   10.0.0.1
ifconfig2  10.0.0.1
root@base:/usr/local/cbsd/tools # cbsd jimport jname=/vagrant/base.img newip=10.0.0.3/24 newjname=ifconfig3
root@base:/usr/local/cbsd/tools # cbsd jls display=jname,ip4_addr | grep ifconfig
ifconfig   10.0.0.1
ifconfig2  10.0.0.1
ifconfig3  10.0.0.1
root@base:/usr/local/cbsd/tools # cbsd jimport jname=/vagrant/base.img newip=dhcp newjname=ifconfig4
root@base:/usr/local/cbsd/tools # cbsd jls display=jname,ip4_addr | grep ifconfig
ifconfig   10.0.0.1
ifconfig2  10.0.0.1
ifconfig3  10.0.0.1
ifconfig4  10.0.0.1
```

**Comment by olevole** [comment]
No, you only need to replace jimport. Put new file into /usr/local/cbsd/jailctl and re-run 
~~~
cbsd initenv
~~~

**Comment by krzysztofantczak** [comment]
Awesome! Thanks! ;-)

**Comment by krzysztofantczak** [comment]
Hey,

Do You have any idea on when can we expect this feature to be available in freebsd cbsd package?

**Comment by olevole** [comment]
Hi,

I plan to look at and fix it in the next two weeks

**Comment by olevole** [comment]
Looks like bug was appeared when I got rid of the old path to rc.conf files (migrating to SQLite3) and this diff https://github.com/olevole/cbsd/commit/3810716090ca474139b9f2217042559f21da070e should fix this error. You can downloaded and apply it via:
```
mv /usr/local/cbsd/tools/imgpart /tmp/imgpart.bak
fetch -o /usr/local/cbsd/tools/imgpart https://raw.githubusercontent.com/olevole/cbsd/master/tools/imgpart
chmod +x /usr/local/cbsd/tools/imgpart
cbsd initenv
````
Unfortunately, at the moment I do not have build farm to re-create and fresh all images on the repository, it will happen for some time

**Comment by krzysztofantczak** [comment]
May I somehow help?

**Comment by olevole** [comment]
No, unfortunately still need my time ;-)

**Comment by olevole** [comment]
Close. Fix will be included in CBSD 10.2.1

**Comment by krzysztofantczak** [comment]
Hey,
Does it mean both IP and jail paths after import are fixed now?
